### PR TITLE
Npt and pressure, volume modifiers

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
This PR integrates the NPT thermostat from ASE.
Further, oscillating temperature and cell volume ramps as well as a pressure ramp have been implemented.

### CODE REVIEW

**Comment by M-R-Schaefer** [comment]
also added stress label histograms and prediction metrics. Both differentiate between hydrostatic and deviatoric stress.

File: ipsuite/calculators/ase_md.py [hunk 1]
```diff
@@ -42,6 +43,55 @@ def modify(self, thermostat, step, total_steps):
...
+    end_cell: int = zntrack.zn.params()
+    cell_amplitude: typing.Union[float, list[float]] = zntrack.zn.params()
+    num_oscillations: float = zntrack.zn.params()
+    interval: int = zntrack.zn.params(1)
+    _initial_cell = None
+
+    def modify(self, thermostat, step, total_steps):
+        if self._initial_cell is None:
+            self._initial_cell = thermostat.atoms.get_cell()
+            if isinstance(self.end_cell, int):
```
Comment (PythonFZ) [inline ipsuite/calculators/ase_md.py:70]: Should this be int or float?

File: ipsuite/calculators/ase_md.py [hunk 2]
```diff
@@ -42,6 +43,55 @@ def modify(self, thermostat, step, total_steps):
...
+    ----------
+    end_cell: float, list[float]
+        cell to ramp to, cubic or tetragonal.
+    cell_amplitude: float
+        amplitude in oscillations of the diagonal cell elements
+    num_oscillations: float
+        number of oscillations. No oscillations will occur if set to 0.
+    interval: int, default 1
+        interval in which the temperature is changed.
+    """
```
Comment (PythonFZ) [inline ipsuite/calculators/ase_md.py:59]: not temperature

File: tests/integration/calculators/test_ase_md.py [hunk 3]
```diff
@@ -37,3 +38,78 @@ def test_ase_md(proj_path, cu_box):
...
+            modifier=[rescale_box],
+            thermostat=thermostat,
+            init_temperature=1.0,
+            steps=20,
+            sampling_rate=1,
+            dump_rate=33,
+        )
+
+    project.run(environment={"OPENBLAS_NUM_THREADS": "1"})
+
```
Comment (PythonFZ) [inline tests/integration/calculators/test_ase_md.py:70]: I don't think we need the env without GAP (same below)

File: tests/integration/calculators/test_ase_md.py [hunk 4]
```diff
@@ -37,3 +38,78 @@ def test_ase_md(proj_path, cu_box):
...
+def test_ase_md_box_ramp(proj_path, cu_box):
+    ase.io.write("cu_box.xyz", cu_box)
+    thermostat = ips.calculators.LangevinThermostat(
+        time_step=1,
+        temperature=1,
+        friction=1,
+    )
+    rescale_box = ips.calculators.BoxOscillatingRampModifier(
+        end_cell=10,
+        cell_amplitude=2,
```
Comment (PythonFZ) [inline tests/integration/calculators/test_ase_md.py:52]: Use a float here instead of an int /(see above)

**Review by PythonFZ — CHANGES_REQUESTED** [review]
This looks really good! I can't wait to try it out! Only a few comments and one issue that has to be resolved.

File: ipsuite/calculators/ase_md.py (hunk 1 above)
Comment (M-R-Schaefer) [reply ipsuite/calculators/ase_md.py:70]: float, fixed

File: tests/integration/calculators/test_ase_md.py (hunk 3 above)
Comment (M-R-Schaefer) [reply tests/integration/calculators/test_ase_md.py:70]: it was adapted from the other test above that also just uses EMT.But good point, I''ll remove it from all of them

File: tests/integration/calculators/test_ase_md.py (hunk 4 above)
Comment (M-R-Schaefer) [reply tests/integration/calculators/test_ase_md.py:52]: good catch!

File: ipsuite/nodes.py [hunk 5]
```diff
@@ -47,6 +47,7 @@ class _Nodes:
     DipoleHistogram = "ipsuite.analysis.DipoleHistogram"
     EnergyHistogram = "ipsuite.analysis.EnergyHistogram"
     ForcesHistogram = "ipsuite.analysis.ForcesHistogram"
+    StressHistogram = "ipsuite.analysis.StressHistogram"
```
Comment (PythonFZ) [inline ipsuite/nodes.py:50]: Are any of the histrogram plots tested?

**Review by PythonFZ — APPROVED** [review]
Looks good! I guess the histogram tests can be done in another PR.

**Pull request merged**